### PR TITLE
refactor: unified factory and informer

### PR DESCRIPTION
**Pull request by AlinsRan**
### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

fix:

With the increase of functional requirements, the interdependence between resources is inevitable. We should unify the informer structure to avoid errors caused by resources not being ready.

image

image
### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: pkg/providers/types/types.go [hunk 1]
```diff
@@ -55,36 +63,58 @@ type ListerInformer struct {
...
+	ApisixRouteLister         kube.ApisixRouteLister
+	ApisixUpstreamLister      kube.ApisixUpstreamLister
+	ApisixPluginConfigLister  kube.ApisixPluginConfigLister
+	ApisixConsumerLister      kube.ApisixConsumerLister
+	ApisixTlsLister           kube.ApisixTlsLister
+	ApisixClusterConfigLister kube.ApisixClusterConfigLister
 }
 
-func (c *ListerInformer) Run(ctx context.Context) {
+func (c *ListerInformer) StartAndWaitSync(ctx context.Context) bool {
```
Comment (tao12345666333) [inline pkg/providers/types/types.go:87]:
```suggestion
func (c *ListerInformer) StartAndWaitForCacheSync(ctx context.Context) bool {
```

Can we rename it?

**Review by lingsamuel — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Comment by tao12345666333** [comment]
I communicated offline with @AlinsRan   who has been testing this PR for a long time.

So I think it's safe to merge it now

**Comment by tao12345666333** [comment]
I think we should add some performance testing, and get related metrics and reports.

**Pull request merged**